config_converter/config_converter.py

```diff
@@ -0,0 +1,202 @@
+"""Program that converts a fluentd config file to a master agent config file.
+
+Usage: Run the exec file in the bin folder with 2 arguments - the
+path to the config file to migrate, and a directory to store the output
+master agent config file in.
+
+To do:
+1. Improve the cli. Include a --help option. Allow optional args like
+   log-level and log-filepath (see project docs).
+2. Create a mapping for log_level if there in the fluentd config file and input
+   default ones if not given
+3. Generate stats like #fields converted, ignored, and output as schema
+4. Generate logs
+5. Fix types of numbers, lists
+6. Generate tests to check if for right logs, right stats, above changes
+"""
+
+import copy
+import sys
+import os
+import subprocess
+import yaml
+from google.protobuf import json_format
+import config_converter.config_pb2 as config_pb2
+
+
+def read_file(path):
```

Comment:
As this is new software, I think we'll use python3 - hence type annotation is available.

Can you add the types annotations for all the functions? I think it helps readability and maintainability, also the type checker can help you identify error at build time.

 https://google.github.io/styleguide/pyguide.html#221-type-annotated-code

Reply:
This is a really cool feature I didn't know about! I've added it in now.